osf/migrations/0045_backfill_date_retracted.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9 on 2017-03-14 14:41
+
+"""
+Backfill Retraction.date_retracted with `RETRACTION_APPROVED` log date.
+"""
+
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+from osf.models import Registration, Retraction, Sanction
+
+
+def forward(*args):
+    retractions = Retraction.objects.filter(state=Sanction.APPROVED, date_retracted=None)
```

Comment:
If you did `Retraction.objects.filter(state=Sanction.APPROVED, date_retracted=None).select_related('registration')` this would be one query instead of having to lazily get the registrations below.